pyomo/contrib/community_detection/detection.py

```diff
@@ -0,0 +1,655 @@
+"""
+Main module for community detection integration with Pyomo models.
+
+This module separates model components (variables, constraints, and objectives) into different communities
+distinguished by the degree of connectivity between community members.
+
+Original implementation developed by Rahul Joglekar in the Grossmann research group.
+
+"""
+from logging import getLogger
+
+from pyomo.common.dependencies import attempt_import
+from pyomo.core import ConcreteModel, ComponentMap, Block, Var, Constraint, Objective, ConstraintList
+from pyomo.core.base.objective import _GeneralObjectiveData
+from pyomo.core.expr.current import identify_variables
+from pyomo.core.expr.visitor import replace_expressions
+from pyomo.contrib.community_detection.community_graph import generate_model_graph
+from pyomo.common.dependencies import networkx as nx
+from pyomo.common.dependencies import matplotlib, matplotlib_available
+from itertools import combinations
+
+import copy
+
+logger = getLogger('pyomo.contrib.community_detection')
+
+# Attempt import of louvain community detection package
+community_louvain, community_louvain_available = attempt_import(
+    'community', error_message="Could not import the 'community' library, available via 'python-louvain' on PyPI.")
+
+# Import matplotlib
+if matplotlib_available:
+    plt = matplotlib.pyplot
```

Comment:
With the recent change to `pyomo.common.dependencies`, you should revert f342de2.  That will prevent matplotlib from being unconditionally imported.

The way the deferred imports work, if you cast the `_available` flag to a `bool` (i.e., in an `if`), then the import is attempted.  So, you want to avoid checking if an optional dependency is available at the module scope (i.e., when the module is imported).  The deferred import system is designed so that in most cases you just grab the deferred module and use it like you normally would - it will be imported the first time it is needed.